python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -112,13 +114,23 @@ def __new__(
             last_materialization=check.opt_inst_param(
                 last_materialization, "last_materialization", EventLogEntry
             ),
+            last_storage_id=check.opt_int_param(last_storage_id, "last_storage_id"),
             last_run_id=check.opt_str_param(last_run_id, "last_run_id"),
             asset_details=check.opt_inst_param(asset_details, "asset_details", AssetDetails),
             cached_status=check.opt_inst_param(
                 cached_status, "cached_status", AssetStatusCacheValue
             ),
         )
 
+    @property
+    def last_materialization_record(self) -> Optional[EventLogRecord]:
```

Comment:
this whole thing is a little weird, and I could be convinced to do this the other way around (just pass in the last_materialization_record directly to this constructor).

The reason I didn't do it this way is that this would require a bit more messing around with the GraphQL Instance stuff for DagsterCloud, but that's not a great reason.